Mathlib/Data/Finsupp/Multiset.lean

```diff
@@ -251,3 +256,46 @@ end Finsupp
 theorem Multiset.toFinsupp_strictMono [DecidableEq ι] : StrictMono (@Multiset.toFinsupp ι _) :=
   (@Finsupp.orderIsoMultiset ι).symm.strictMono
 #align multiset.to_finsupp_strict_mono Multiset.toFinsupp_strictMono
+
+namespace Sym
+
+variable (α)
+variable [DecidableEq α] (n : ℕ)
+
+/-- The `n`th symmetric power of a type `α` is naturally equivalent to the subtype of
+finitely-supported maps `α →₀ ℕ` with total mass `n`.
+
+See also `Sym.equivNatSumOfFintype` when `α` is finite. -/
+def equivNatSum :
+    Sym α n ≃ {P : α →₀ ℕ // P.sum (fun _ ↦ id) = n} :=
+  (Multiset.toFinsupp.toEquiv.image _).trans <| Equiv.Set.ofEq <| by
+  ext P
+  refine ⟨?_, ?_⟩
+  · rintro ⟨s, rfl, rfl⟩
+    simp [Set.mem_def]
+  · rintro rfl
+    exact ⟨Multiset.toFinsupp.symm P, by simp [Set.mem_def], by simp⟩
+
+@[simp] lemma coe_equivNatSum_apply_apply (s : Sym α n) (a : α) :
+    (equivNatSum α n s : α →₀ ℕ) a = (s : Multiset α).count a :=
+  rfl
+
+/-- The `n`th symmetric power of a finite type `α` is naturally equivalent to the subtype of maps
+`α → ℕ` with total mass `n`.
+
+See also `Sym.equivNatSum` when `α` is not necessarily finite. -/
+noncomputable def equivNatSumOfFintype [Fintype α] :
+    Sym α n ≃ {P : α → ℕ // ∑ i, P i = n} :=
+  (equivNatSum α n).trans <| (Finsupp.equivFunOnFinite.image _).trans <| Equiv.Set.ofEq <| by
```

Comment:
If you pass this through `DFinsupp.equivFunOnFintype` and `Finsupp.toDFinsupp`, then it would be computable, which is nice for `#eval`

Reply:
If it could be rendered computable for free I would do it, but I think even the low price of having to depend on `DFinsupp.equivFunOnFintype` (incidentally not imported in this file) makes it not worth it for me.

Reply:
Perhaps the better resolution here is to have a `Multiset.countEquiv` or similar that is the `Fintype` analog to `Multiset.toFinsupp`, with a nice defeq for the summation. I'll let you decide if that sounds out of scope

bors d+

Reply:
I guess we should probably have this somewhere:
```lean
def Multiset.countAddEquiv (α : Type*) [Fintype α] [DecidableEq α] :
    Multiset α ≃+ (α → ℕ) where
  toFun s := s.count
  invFun P := ∑ a, P a • {a}
  map_add' s t := by ext; simp
  left_inv s := by
    ext a
    change count a (sum _) = _
    rw [count_sum]
    simp [count_singleton]
  right_inv P := by
    ext a
    change count a (sum _) = _
    rw [count_sum]
    simp [count_singleton]
```
I'll try to come back to this.